src/components/item/item-sliding.ts

```diff
@@ -1,8 +1,42 @@
-import {Component, ElementRef, Optional, ChangeDetectionStrategy, ViewEncapsulation} from '@angular/core';
+import {Component, Directive, Input, Renderer, ContentChildren, ContentChild, ElementRef, Optional, ChangeDetectionStrategy, ViewEncapsulation} from '@angular/core';
+import {NgClass} from '@angular/common';
 
 import {List} from '../list/list';
+import {Item} from './item';
+import {isPresent} from '../../util/util';
+import {CSS} from '../../util/dom';
 
 
+export const enum SideFlags {
+  None = 0,
+  Left = 1 << 0,
+  Right = 1 << 1,
+  Both = Left | Right
+}
+
+@Directive({
+  selector: 'ion-item-options',
+})
+export class ItemOptions {
+  @Input() side: string;
+
+  constructor(private ele: ElementRef) { }
+
+  getSides(): SideFlags {
+    if (!isPresent(this.side) || this.side === 'right') {
+      return SideFlags.Right;
+    } else if (this.side === 'left') {
+      return SideFlags.Left;
+    }
+    console.error("invalid side attribute", this.side);
+    return SideFlags.Right;
+  }
+
+  getWidth(): number {
```

Comment:
This value should be cached so it only needs to be looked up once, otherwise we could have repaint issues. https://github.com/driftyco/ionic/blob/2.0/src/util/dom.ts#L264